src/OpenTelemetry.Abstractions/Trace/ITracer.cs

```diff
@@ -50,107 +51,71 @@ public interface ITracer
         // TODO: add sampling hints
 
         /// <summary>
-        /// Creates root span.
-        /// </summary>
-        /// <param name="operationName">Span name.</param>
-        /// <param name="kind">Kind.</param>
-        /// <param name="startTimestamp">Start timestamp.</param>
-        /// <param name="linksGetter">Func to retrieve the Links collection.</param>
-        /// <returns>Span instance.</returns>
-        ISpan StartRootSpan(string operationName, SpanKind kind, DateTimeOffset startTimestamp, Func<IEnumerable<Link>> linksGetter);
-
-        /// <summary>
-        /// Creates root span.
-        /// </summary>
-        /// <param name="operationName">Span name.</param>
-        /// <param name="kind">Kind.</param>
-        /// <param name="startTimestamp">Start timestamp.</param>
-        /// <param name="links">collection.</param>
-        /// <returns>Span instance.</returns>
-        ISpan StartRootSpan(string operationName, SpanKind kind, DateTimeOffset startTimestamp, IEnumerable<Link> links);
-
-        /// <summary>
-        /// Creates span. If there is active current span, it becomes a parent for returned span.
+        /// Starts root span.
         /// </summary>
         /// <param name="operationName">Span name.</param>
         /// <param name="kind">Kind.</param>
-        /// <param name="startTimestamp">Start timestamp.</param>
-        /// <param name="linksGetter">Func to retrieve the Links collection.</param>
+        /// <param name="options">Advanced span creation options.</param>
         /// <returns>Span instance.</returns>
-        ISpan StartSpan(string operationName, SpanKind kind, DateTimeOffset startTimestamp, Func<IEnumerable<Link>> linksGetter);
+        ISpan StartRootSpan(string operationName, SpanKind kind, SpanCreationOptions options);
 
         /// <summary>
-        /// Creates span. If there is active current span, it becomes a parent for returned span.
+        /// Starts activate root span.
         /// </summary>
         /// <param name="operationName">Span name.</param>
         /// <param name="kind">Kind.</param>
-        /// <param name="startTimestamp">Start timestamp.</param>
-        /// <param name="links">Links collection.</param>
+        /// <param name="options">Advanced span creation options.</param>
         /// <returns>Span instance.</returns>
-        ISpan StartSpan(string operationName, SpanKind kind, DateTimeOffset startTimestamp, IEnumerable<Link> links);
+        IDisposable StartActiveRootSpan(string operationName, SpanKind kind, SpanCreationOptions options);
```

Comment:
Also if we adding it - would be nice to have an option to get the Span as an output argument so you can add attributes to span inside the `using` block without the need to get it from `Tracer.Current`

Reply:
You can cast output to span. Another option is to make ISpan disposable and simply return ISpan.

Reply:
Would it be better to return an `ISpan` as a out argument?

Reply:
the point is that you need disposable to do `using (StartActiveSpan)`. I'm playing with making ISpan disposable,  that would allow that. Do you have any objections agains disposable ISpan

Reply:
On a first though Disposable Span sounds scary. Need to think more.

The point of out parameter is just a quick access to the span. It is just a perf optimization.

``` csharp
using (var scope = tracer.StartActiveSpan("scoped span", out var span))
{
  if (span.IsRecording)
    span.AddAttribute("new", "attribute");
}
```

I don't see how it will create a confusion in that use. The worst scenario is this:

``` csharp
callback.OnBegin( () => {
  var scope = tracer.StartActiveSpan("scoped span", out var span);
} );
callback.OnEnd( () => {
  span = Tracer.Current;
  span.End(); // <-- will this restore to the parent's scope? Probably not. So how to get a scope here?
} );


```

Reply:
```
callback.OnBegin( () => {
  var scope = tracer.StartActiveSpan("scoped span", out var span);
} );
callback.OnEnd( () => {
  span = Tracer.Current;
  span.End(); // <-- will this restore to the parent's scope? Probably not. So how to get a scope here?
} );
```

this is one of the problems I wanted to tackle. `scope` is left undisposed. End() should not end the scope and I saw people using concurrent dictionaries on many occasions to solve this problem.

Making span disposable solves it. But yes, sounds a bit scary.

Reply:
And this problem exists without convenience methods too

Reply:
I see just a couple of ways to fix it:
1. Span has scope property. Then you can get Scope from current span. When span is disposed, it disposes scope. You could end span before that
2. Scope has span property - then we get CurrentScope (not current Span) and we have yet another type for it
3. Span is the scope (this PR) - then there is also additional scope returned by WithSpan so you can control scope lifetime separately from span

All of the above works only when we make changes on ISpan, cannot only change implementation. I kinda like p1. What do you think?